tests/app/test_jinja_filters.py

```diff
@@ -460,3 +375,106 @@ def test_format_address_fields_with_uprn():
     )
+
+
+def _get_test_field_from_answer_schema(answer_schema):
+    questionnaire_schema = Mock()
+    questionnaire_schema.error_messages = error_messages
+    question_schema = {"answers": [answer_schema]}
+    form = generate_form(
+        questionnaire_schema,
+        question_schema,
+        answer_store=Mock(),
+        metadata={},
+        location=None,
+        data=None,
+        form_data=None,
+    )
+
+    return getattr(form, answer_schema["id"])
+
+
+@pytest.mark.parametrize(
+    "min_or_max_value, expected_width",
+    [
+        (None, 10),
+        (1, 1),
+        (123123123123, 20),
+    ],
+)
+def test_numeric_detail_answer_input_class_config(
+    answer_schema_number, min_or_max_value, expected_width
+):
+    if min_or_max_value:
+        answer_schema_number["maximum"] = {"value": min_or_max_value}
+
+    other = OtherConfig(Mock(), answer_schema_number)
+    assert other.classes == f"input--w-{expected_width}"
+
+
+def _test_detail_answer_config(answer_schema, is_visible, expected_visibility):
+    if is_visible is True:
+        answer_schema["visible"] = True
+    elif is_visible is False:
+        answer_schema["visible"] = False
+
+    field = _get_test_field_from_answer_schema(answer_schema=answer_schema)
+
+    other = OtherConfig(field, answer_schema)
+
+    if answer_schema["type"] == "Dropdown":
+        expected_type = "select"
+        assert hasattr(other, "options")
+        assert not hasattr(other, "value")
+    else:
+        expected_type = "input"
+        assert not hasattr(other, "options")
+        assert hasattr(other, "value")
+
+    assert other.id == answer_schema["id"]
+    assert other.name == answer_schema["id"]
+    assert other.label.text == answer_schema["label"]
+    assert other.otherType == expected_type
+    assert other.open is expected_visibility
+
+
+@pytest.mark.parametrize(
+    "is_visible, expected_visibility",
+    [
+        (True, True),
+        (False, False),
+        (None, False),
+    ],
+)
+def test_numeric_detail_answer_config(
+    app, answer_schema_number, is_visible, expected_visibility
+):
+    _test_detail_answer_config(answer_schema_number, is_visible, expected_visibility)
+
+
+@pytest.mark.parametrize(
+    "is_visible, expected_visibility",
+    [
+        (True, True),
+        (False, False),
+        (None, False),
+    ],
+)
+def test_textfield_detail_answer_config(
+    app, answer_schema_textfield, is_visible, expected_visibility
+):
+    _test_detail_answer_config(answer_schema_textfield, is_visible, expected_visibility)
+
+
+@pytest.mark.parametrize(
+    "is_visible, expected_visibility",
+    [
+        (True, True),
+        (False, False),
+        (None, False),
+    ],
+)
+def test_dropdown_detail_answer_config(
+    app, answer_schema_dropdown, is_visible, expected_visibility
+):
+    _test_detail_answer_config(answer_schema_dropdown, is_visible, expected_visibility)
```

Comment:
I find these tests hard to follow. They are parametrized to test for visibility, but test all properties of `OtherConfig`. 

I think the tests for visibility should be separated from the tests for the detail answer types. Also, as the visibility logic is the same for all detail answer types, we should only need to test one type

Reply:
I'll make 1 test for visibility and 1 tests for the input type check.

Reply:
Have pushed some tests that should be easier to digest.